### PR TITLE
Window abstraction cleanups

### PR DESCRIPTION
**Pull request by tronical**
I need to do more testing (results are looking okay at first glance), but code wise I think that this is ready for review.

### CODE REVIEW

File: sixtyfps_runtime/corelib/window.rs [hunk 1]
```diff
@@ -103,6 +105,23 @@ impl crate::properties::PropertyChangeHandler for WindowRedrawTracker {
     }
 }
 
+/// This enum describes the different ways a popup can be rendered by the back-end.
+pub enum PopupWindowLocation {
+    /// The popup is rendered in its own top-level window that is know to the windowing system.
+    TopLevel(Rc<Window>),
+    /// The popup is rendered as an embedded child window at the given position.
+    ChildWindow(Point),
```
Comment (ogoffart) [inline sixtyfps_runtime/corelib/window.rs:113]: I'm a bit torn there.
Even if the backend don't render as a separate window (which the GL backend eventually should do anyway), should there not be, conceptually, a different Rc<Window> for the popup anyway?

File: sixtyfps_runtime/rendering_backends/gl/graphics_window.rs [hunk 2]
```diff
@@ -218,15 +124,21 @@ impl GraphicsWindow {
             }
         };
 
-        let backend = self.window_factory.as_ref()(window_builder);
+        let backend = crate::GLRenderer::new(
```
Comment (ogoffart) [inline sixtyfps_runtime/rendering_backends/gl/graphics_window.rs:127]: Is `backend` the right name here?

Do we even need GLRenderer at all

File: sixtyfps_runtime/corelib/window.rs (hunk 1 above)
Comment (tronical) [reply sixtyfps_runtime/corelib/window.rs:113]: In torn, too. It’s tempting but otoh it means there will be a circular dependency. So if the application code decides to drop its ref to the window while a pop up is active, then I think the window will leak?

File: sixtyfps_runtime/rendering_backends/gl/graphics_window.rs (hunk 2 above)
Comment (tronical) [reply sixtyfps_runtime/rendering_backends/gl/graphics_window.rs:127]: We do not, that’s right. I wanted to continue to with that but the patch become a bit large :-)

**Review by ogoffart — APPROVED** [review]

File: sixtyfps_runtime/corelib/window.rs [hunk 3]
```diff
@@ -103,6 +105,23 @@ impl crate::properties::PropertyChangeHandler for WindowRedrawTracker {
...
+pub struct PopupWindow {
+    /// The location defines where the pop up is rendered.
+    pub location: PopupWindowLocation,
+    /// The component that is responsible for providing the popup content.
+    pub component: ComponentRc,
+}
+
 /// Structure that represent a Window in the runtime
 pub struct Window {
     /// FIXME! use Box instead;
```
[the remark this answers is not in the document]
Comment (tronical) [reply sixtyfps_runtime/corelib/window.rs:127]: I'll look into that.

[the remark this answers is not in the document]
Comment (tronical) [reply sixtyfps_runtime/corelib/window.rs:127]: I tried that locally and it's possible to remove the Rc from the platform window. Some parts become a little messier, some cleaner.

**Pull request merged**